### PR TITLE
wrf: add v4.5.0

### PR DESCRIPTION
**Pull request by MichaelLaufer**
- Patch needed for new version of Adios2, similar to here: https://github.com/spack/spack/pull/36563/
- Upstream PR submitted: https://github.com/wrf-model/WRF/pull/1860

### CODE REVIEW

File: var/spack/repos/builtin/packages/wrf/package.py [hunk 1]
```diff
@@ -157,7 +163,11 @@ class Wrf(Package):
     patch("patches/4.2/derf_fix.patch", when="@4.2 %aocc")
 
     patch("patches/4.4/arch.postamble.patch", when="@4.4:")
-    patch("patches/4.4/configure.patch", when="@4.4:")
+    patch("patches/4.4/configure.patch", when="@4.4:4.4.2")
+
+    patch("patches/4.5/configure.patch", when="@4.5:")
+    patch("patches/4.5/adios2-remove-deprecated-functions.patch", when="@4.5: ^adios2@2.9:")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/wrf/package.py]: Can you add brief comments explaining what is the effect of the patch, why it is needed and where it was taken from?

File: var/spack/repos/builtin/packages/wrf/package.py [hunk 2]
```diff
@@ -205,6 +215,7 @@ class Wrf(Package):
     depends_on("time", type=("build"))
     depends_on("m4", type="build")
     depends_on("libtool", type="build")
+    depends_on("adios2", when="+adios2")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/wrf/package.py]: Does this work also for older versions?

**Review by alalazo — CHANGES_REQUESTED** [review]
A couple of comments

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 2 above)
Comment (MichaelLaufer) [reply var/spack/repos/builtin/packages/wrf/package.py]: Good catch. Done.

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 1 above)
Comment (MichaelLaufer) [reply var/spack/repos/builtin/packages/wrf/package.py]: Done.

**Review by alalazo — APPROVED** [review]

**Pull request merged**